### PR TITLE
AU-666: Delete bank confirmation files automatically

### PR DESCRIPTION
**Pull request by Krisseck**
# [AU-666](https://helsinkisolutionoffice.atlassian.net/browse/AU-666)

Previously application form did not remove bank account confirmation files when bank account was changed on the application. This fixes it.

## What was done

Bank account confirmation files are automatically deleted from ATV on profile page.

Also they are deleted on application page.

Also grants profiles were set up to use different profiles per environment. This helps with integrationID's on files and is not bad idea in general

As of 1625 on 15.3 this is installed in DEV environment and can be tested there also.

## How to install

* Checkout
* Make fresh
* Drush cr

## How to test

* [ ] Add a bank account and confirmation file to profile
* [ ] Delete said bank account
* [ ] Check logs that ATV file deletion succeeded
* [ ] Create new application
* [ ] Select bank account, save
* [ ] Open form again, change bank account, see from logs that the file got deleted
 Look for something like this in drupal logs:
`ATV DELETE query https://atv-api-hki-kanslia-atv-test.agw.arodevtest.hel.fi/v1/documents/f5d28c7e-4c81-42bf-8ebe-402cb50c9dff/attachments/4898/ took 241 ms`

[AU-666]: https://helsinkisolutionoffice.atlassian.net/browse/AU-666?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jiisuominen** [comment]
Oh, apologies. This should also fix the bank account change on application. Updated the ticket as well.

Sorry.

**Review by jiisuominen — CHANGES_REQUESTED** [review]
Sorry about bad specs.

**Review by jiisuominen — DISMISSED** [review]
Toimiiks tää jo? Ei committeja moniin päiviin?

**Comment by Krisseck** [comment]
No ei varsinaisesti toimi, haluan testata sen kunnolla, mutta en saa vielä VPN:ää toimimaan 

On March 13, 2023 7:51:24 AM GMT+02:00, Janne Suominen ***@***.***> wrote:
>@jiisuominen requested changes on this pull request.
>
>Toimiiks tää jo? Ei committeja moniin päiviin?
>
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/City-of-Helsinki/hel-fi-drupal-grants/pull/192#pullrequestreview-1336281511
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by Puavo** [comment]
Mä aloitin perjantaina katselemaan tätä. SonarCloud valittelee että sisennyksiä on liikaa.

Omana kommenttina renderöinti komennon elseif:in ehdossa on aika raskasta luettavaa. Mä korjailen näitä.

File: public/modules/custom/grants_attachments/src/AttachmentHandler.php [hunk 1]
```diff
@@ -555,26 +587,71 @@ public function handleBankAccountConfirmation(
...
+              $isAccountConfirmation = FALSE;
+              $integrationID = NULL;
+
+              foreach ($attachmentArray as $attachmentItemArray) {
+                if ($attachmentItemArray["ID"] === "integrationID") {
+                  $integrationID = $attachmentItemArray["value"];
+                }
+                elseif (
+                  $attachmentItemArray["ID"] === "description" &&
+                  str_contains($attachment['value'], t('Confirmation for account @accountNumber', ['@accountNumber' => ''])->render())
```
Comment (jiisuominen) [inline public/modules/custom/grants_attachments/src/AttachmentHandler.php:614]: Joo, eli ongelma on se, että tässä testataan tota stringin sisältöä, joka on hieman arveluttava tapa.

Parempi tapa selvittää onko kyseessä liitteen validointitiedosto on tutkia fileType koodia. Pankkitilin vahvistusliitteet ovat fileType = 45

**Comment by Puavo** [comment]
Koodit näyttävät nyt monta pykälää selkeämmältä.

**Review by Puavo — APPROVED** [review]
Hyvältä vaikuttaa.

**Pull request merged**